src/highdicom/map/sop.py

```diff
@@ -0,0 +1,789 @@
+            A tuple where the first element is the enum value and the second
+            value is the DICOM pixel data attribute for the given datatype.
+            One of (``"PixelData"``, ``"FloatPixelData"``,
+            ``"DoubleFloatPixelData"``)
+
+        Raises
+        ------
+        ValueError
+            If values in the input array don't have a supported unsigned
+            integer or floating-point type.
+
+        """
+        if pixel_array.dtype.kind == 'f':
+            # Further check for float32 vs float64
+            if pixel_array.dtype.name == 'float32':
+                return (
+                    _PixelDataType.SINGLE,
+                    self._pixel_data_type_map[_PixelDataType.SINGLE],
+                )
+            elif pixel_array.dtype.name == 'float64':
+                return (
+                    _PixelDataType.DOUBLE,
+                    self._pixel_data_type_map[_PixelDataType.DOUBLE],
+                )
+            else:
+                raise ValueError(
+                    'Unsupported floating-point type for pixel data: '
+                    '32-bit (single-precision) or 64-bit (double-precision) '
+                    'floating-point types are supported.'
+                )
+        elif pixel_array.dtype.kind == 'u':
+            if pixel_array.dtype not in (np.uint8, np.uint16):
+                raise ValueError(
+                    'Unsupported unsigned integer type for pixel data: '
+                    '16-bit unsigned integer types are supported.'
+                )
+            return (
+                _PixelDataType.USHORT,
+                self._pixel_data_type_map[_PixelDataType.USHORT],
+            )
+        elif pixel_array.dtype.kind == "i":
+            if pixel_array.dtype not in (np.int8, np.int16):
+                raise ValueError(
+                    'Unsupported signed integer type for pixel data: '
+                    '8-bit or 16-bit signed integer types are supported.'
+                )
+            return (
+                _PixelDataType.SHORT,
+                self._pixel_data_type_map[_PixelDataType.SHORT],
+            )
+        raise ValueError(
+            'Unsupported data type for pixel data.'
+            'Supported are 8-bit or 16-bit signed and unsigned integer types '
+            'as well as 32-bit (single-precision) or 64-bit (double-precision) '
+            'floating-point types.'
+        )
+
+    def _encode_pixels(self, plane: np.ndarray) -> bytes:
+        """Encodes a given pixel array as a bytes object
+
+        Parameters
+        ----------
+        plane : np.ndarray
+            The numpy array to encode
+
+        Returns
+        -------
+        bytes
+            `plane` encoded as a `bytes` object
+
+        Raises
+        ------
+        ValueError
+            If the SOP instance uses an encapsulated transfer syntax and
+            `plane` is not exactly 2 dimensional.
+        """
```

Comment:
```suggestion
            `plane` is not exactly 2 dimensional.

        """
```

Reply:
Closing this comment, since it's outdated.